### ISSUE TITLE
[Android] Can't load any ETC1 textures

### ISSUE THREAD

**Issue by mrhelmut** [MonoGame/MonoGame#5984]
On latest develop, I can't get any ETC1 compressed textures to load.

If there is no mipmap, it fails here: https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Content/ContentReaders/Texture2DReader.cs#L74

... with a "can't read behind end of stream" error (levelCount, height and width are read correctly).

If there are mipmaps, it fails here: https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/Texture2D.cs#L377

I think it's related to #5983 (or possibly its cause/consequence).

@dellis1972 would you have any suggestion on where to look? (I believe it happened on late OpenTK versions too.)

**Comment by dellis1972** [comment]
@mrhelmut I think @KonajuGames will need to take a look at this one.

**Comment by mrhelmut** [comment]
I think it has something to do the data size.
On the OpenTK version, it fails ```CompressedTexSubImage2D```, and the the ```elementCount``` is different than the ```dataPtr``` lenght.

**Comment by mrhelmut** [comment]
Setting the data with ```CompressedTexImage2D``` instead of ```CompressedTexSubImage2D``` works. Most probably a parameter issue.

**Comment by mrhelmut** [comment]
Confirmed, it works on another Android device.

Is there any way to fallback to ```CompressedTexImage2D``` when ```rect``` is the full texture?

**Comment by Jjagg** [comment]
> Is there any way to fallback to CompressedTexImage2D when rect is the full texture?

Yeah, that would be better. That's already done when the overload without a rect is called, but if you use the overload with a rectangle and pass null for it or the rect for the full texture, it does a `*TexSubImage2D` call.

**Comment by KonajuGames** [comment]
Fixing that right now along with falling back to 16-bit textures when not able to process into ETC1. Also found some other issues, like `ContentLoadException` being silently consumed and results in the app stuck in an endless loop of trying to load content and failing.

**Comment by KonajuGames** [comment]
Writing a bunch of unit tests for this.  It's after 3am now, so I have to get some sleep. Will finish this tomorrow night.

**Comment by mrhelmut** [comment]
Please have some rest! Thank you for taking a look at this!
